### PR TITLE
website/integrations: Updated AWS docs for the new IAM Center and SCIM

### PR DESCRIPTION
**Pull request by justsem**
Updated the AWS Integration docs to match the new IAM Centre's method. This includes SCIM.

## Changes

The AWS integration docs have been updated to reflect the new IAM Identity Center.
This also adds configuration guidance for SCIM.

-   [X] The documentation has been updated
-   [X] The documentation has been formatted (`make website`) 
     (I believe the CI takes care of this itself, right?)

### CODE REVIEW

**Comment by BeryJu** [comment]
Thanks for the PR! The traditional SAML authentication with AWS is still useful in some cases so I'd prefer to keep that around and add additional docs for the AWS IAM center/AWS SSO

**Comment by justSem** [comment]
Thanks for your feedback.
I've updated the docs with 2 sections, each describing a way to set up the integration

File: website/integrations/services/aws/index.md [hunk 1]
```diff
@@ -75,3 +82,94 @@ To use the user's username, use this snippet
...
+## Preparation
+
+The following placeholders will be used:
+
+-   `authentik.company` is the FQDN of the authentik install.
+
+Additional Preparation:
+
+-   A certificate to sign SAML assertions is required. You can use authentik's default certificate, or provide/generate one yourself.
+-   You may pre-create an AWS application
```
[the remark this answers is not in the document]
Comment (tanberry) [reply website/integrations/services/aws/index.md:97]: "may" or "need to" or... using the word "may" is a little unclear here, not sure if I need to or not. If it is optional, we can say:

"Optionally, you can pre-create an AWS application."

File: website/integrations/services/aws/index.md [hunk 2]
```diff
@@ -75,3 +82,94 @@ To use the user's username, use this snippet
...
+
+## Procedure
```
Comment (tanberry) [inline website/integrations/services/aws/index.md:99]:
```suggestion
## How to integrate with AWS
```

File: website/integrations/services/aws/index.md [hunk 3]
```diff
@@ -75,3 +82,94 @@ To use the user's username, use this snippet
...
+
+In AWS:
+
+-   In AWS Navigate to: `IAM Identity Center -> Settings -> Identity Source (tab)`
```
Comment (tanberry) [inline website/integrations/services/aws/index.md:103]:
```suggestion
-   In AWS navigate to: `IAM Identity Center -> Settings -> Identity Source (tab)`
```

File: website/integrations/services/aws/index.md [hunk 4]
```diff
@@ -75,3 +82,94 @@ To use the user's username, use this snippet
...
+-   On the right hand side click `Actions -> Change identity source`
```
Comment (tanberry) [inline website/integrations/services/aws/index.md:104]:
```suggestion
-   On the right side click `Actions -> Change identity source`
```

File: website/integrations/services/aws/index.md [hunk 5]
```diff
@@ -75,3 +82,94 @@ To use the user's username, use this snippet
...
+-   Select `External Identity Provider`
+-   Under `Service Provider metadata` it'll allow you to download a metadata file. Download this file.
```
Comment (tanberry) [inline website/integrations/services/aws/index.md:106]:
```suggestion
-   Under `Service Provider metadata` download the metadata file. 
```

File: website/integrations/services/aws/index.md [hunk 6]
```diff
@@ -75,3 +82,94 @@ To use the user's username, use this snippet
...
+
+Now go to your authentik instance, and perform the following steps.
+
+-   Under _Providers_ create a new _SAML Provider from metadata_. Give it a name, and upload the metadata file AWS gave you.
+-   Click _Next_. Give it a name, and close.
+-   If you haven't done so yet, create an application for AWS and connect the Provider to it.
```
Comment (tanberry) [inline website/integrations/services/aws/index.md:112]:
```suggestion
-   If you haven't done so yet, create an application for AWS and connect the provider to it.
```

Comment (tanberry) [reply website/integrations/services/aws/index.md:112]: ideally we would be able to link to a procedural doc on how to create an app... but we will add that later.

File: website/integrations/services/aws/index.md [hunk 7]
```diff
@@ -75,3 +82,94 @@ To use the user's username, use this snippet
...
+-   Navigate to the provider you've just created, select _Edit_
```
Comment (tanberry) [inline website/integrations/services/aws/index.md:113]:
```suggestion
-   Navigate to the provider you've just created, and then select _Edit_
```

File: website/integrations/services/aws/index.md [hunk 8]
```diff
@@ -75,3 +82,94 @@ To use the user's username, use this snippet
...
+
+-   In AWS Navigate to: `IAM Identity Center -> Settings -> Identity Source (tab)`
+-   On the right hand side click `Actions -> Change identity source`
+-   Select `External Identity Provider`
+-   Under `Service Provider metadata` it'll allow you to download a metadata file. Download this file.
+
+Now go to your authentik instance, and perform the following steps.
+
+-   Under _Providers_ create a new _SAML Provider from metadata_. Give it a name, and upload the metadata file AWS gave you.
+-   Click _Next_. Give it a name, and close.
```
Comment (tanberry) [inline website/integrations/services/aws/index.md:111]:
```suggestion
-   Click _Next_. Give it a name, and close the file.
```

File: website/integrations/services/aws/index.md [hunk 9]
```diff
@@ -75,3 +82,94 @@ To use the user's username, use this snippet
...
+-   If you haven't done so yet, create an application for AWS and connect the Provider to it.
+-   Navigate to the provider you've just created, select _Edit_
+-   Copy the _Issuer URL_ to the _Audience_ field.
+-   Under _Advanced Protocol Settings_ set a _Signing Certificate_
+-   Save and Close.
+-   Under _Related Objects_ download the _Metadata file_, and the _Signing Certificate_
+
+Now go back to your AWS instance
+
+-   Under `Identity provider metadata` upload both the the `Metadata file` and `Signing Certificate` authentik gave you.
```
Comment (tanberry) [inline website/integrations/services/aws/index.md:121]:
```suggestion
-   Under `Identity provider metadata` upload both the the `Metadata` file and `Signing Certificate` that authentik gave you.
```

File: website/integrations/services/aws/index.md [hunk 10]
```diff
@@ -75,3 +82,94 @@ To use the user's username, use this snippet
...
+-   Choose `Next`.
```
Comment (tanberry) [inline website/integrations/services/aws/index.md:122]:
```suggestion
-   Click `Next`.
```

File: website/integrations/services/aws/index.md [hunk 11]
```diff
@@ -75,3 +82,94 @@ To use the user's username, use this snippet
...
+-   In your settings pane, under the tab `Identity Source` click `Actions -> Manage Authentication`
```
Comment (tanberry) [inline website/integrations/services/aws/index.md:123]:
```suggestion
-   In your settings pane, under the tab `Identity Source`, click `Actions -> Manage Authentication`.
```

File: website/integrations/services/aws/index.md [hunk 12]
```diff
@@ -75,3 +82,94 @@ To use the user's username, use this snippet
...
+-   Take note of the `AWS access portal sign-in URL` (this is especially important if you've changed it from the default)
```
Comment (tanberry) [inline website/integrations/services/aws/index.md:124]:
```suggestion
-   Take note of the `AWS access portal sign-in URL` (this is especially important if you changed it from the default).
```

File: website/integrations/services/aws/index.md [hunk 13]
```diff
@@ -75,3 +82,94 @@ To use the user's username, use this snippet
...
+
+Now go back to your authentik instance.
+
+-   Navigate to the Application you've created for AWS and click _Edit_.
```
Comment (tanberry) [inline website/integrations/services/aws/index.md:128]:
```suggestion
-   Navigate to the Application that you created for AWS and click _Edit_.
```

File: website/integrations/services/aws/index.md [hunk 14]
```diff
@@ -75,3 +82,94 @@ To use the user's username, use this snippet
...
+-   Under _UI Settings_ make sure the _Start URL_ matches the _AWS access portal sign-in URL_
+
+## Caveats and Troubleshooting
+
+-   Users need to exist in AWS in order to use them through authentik. AWS will throw an error if it doesn't recognise the user.
```
Comment (tanberry) [inline website/integrations/services/aws/index.md:133]:
```suggestion
-   Users need to already exist in AWS in order to use them through authentik. AWS will throw an error if it doesn't recognise the user.
```

File: website/integrations/services/aws/index.md [hunk 15]
```diff
@@ -75,3 +82,94 @@ To use the user's username, use this snippet
...
+-   In case you're stuck, you can see the SSO logs in Amazon CloudTrail -> Event History. Look for `ExtenalIdPDirectoryLogin`
+
+Note:
+
+## Optional: Auto provisioning with SCIM
```
Comment (tanberry) [inline website/integrations/services/aws/index.md:138]:
```suggestion
## Optional: Automated provisioning with SCIM
```

File: website/integrations/services/aws/index.md [hunk 16]
```diff
@@ -75,3 +82,94 @@ To use the user's username, use this snippet
...
+
+Some people may opt for the automatic provisioning feature called SCIM.
```
Comment (tanberry) [inline website/integrations/services/aws/index.md:140]:
```suggestion
Some people may opt TO USE the automatic provisioning feature called SCIM (System for Cross-domain Identity Management).
```

File: website/integrations/services/aws/index.md [hunk 17]
```diff
@@ -75,3 +82,94 @@ To use the user's username, use this snippet
...
+SCIM allows you to synchronise (part of) your directory to AWS's IAM. Saving you the hassle of having to create users by hand.
```
Comment (tanberry) [inline website/integrations/services/aws/index.md:141]:
```suggestion
SCIM allows you to synchronize (part of) your directory to AWS's IAM, saving you the hassle of having to create users by hand.
```

File: website/integrations/services/aws/index.md [hunk 18]
```diff
@@ -75,3 +82,94 @@ To use the user's username, use this snippet
...
+In order to do so, go to your AWS Identity Center
```
Comment (tanberry) [inline website/integrations/services/aws/index.md:142]:
```suggestion
In order to do so, take the following steps in your AWS Identity Center:
```

File: website/integrations/services/aws/index.md [hunk 19]
```diff
@@ -75,3 +82,94 @@ To use the user's username, use this snippet
...
+
+-   In your `Settings` pane, locate the `Automatic Provisioning` Info box. Click `Enable`
```
Comment (tanberry) [inline website/integrations/services/aws/index.md:144]:
```suggestion
-   In your `Settings` pane, locate the `Automatic Provisioning` Info box. Click `Enable`.
```

File: website/integrations/services/aws/index.md [hunk 20]
```diff
@@ -75,3 +82,94 @@ To use the user's username, use this snippet
...
+-   AWS will give you an `SCIM Endpoint` and a `Access Token`. Take note of these
```
Comment (tanberry) [inline website/integrations/services/aws/index.md:145]:
```suggestion
-   AWS will give you an `SCIM Endpoint` and a `Access Token`. Take note of these values.
```

File: website/integrations/services/aws/index.md [hunk 21]
```diff
@@ -75,3 +82,94 @@ To use the user's username, use this snippet
...
+
+Go back to your authentik instance
+
+-   Navigate to _Providers_ -> _Create_
+-   Select _SCIM Provider_
+-   Give it a name, under _URL_ enter the _SCIM Endpoint_ and under _Token_ enter the _Access Token_ AWS provided you with.
```
Comment (tanberry) [inline website/integrations/services/aws/index.md:151]:
```suggestion
-   Give it a name, under _URL_ enter the _SCIM Endpoint_, and then under _Token_ enter the _Access Token_ AWS provided you with.
```

File: website/integrations/services/aws/index.md [hunk 22]
```diff
@@ -75,3 +82,94 @@ To use the user's username, use this snippet
...
+-   In case you wish, change the user filtering settings to your liking. Click _Finish_
```
Comment (tanberry) [inline website/integrations/services/aws/index.md:152]:
```suggestion
-   Optionally, change the user filtering settings to your liking. Click _Finish_
```

File: website/integrations/services/aws/index.md [hunk 23]
```diff
@@ -75,3 +82,94 @@ To use the user's username, use this snippet
...
+
+-   Go to _Customization -> Property Mappings_
+-   Click _Create -> SCIM Mapping_
+-   Make sure to give the mapping a name that's lexically lower than `authentik default`, for example `AWS SCIM User mapping`
+-   As the expression, enter:
+
+```python
+# The only thing this does, is strip the default mapping from it's 'photos' attribute,
```
Comment (tanberry) [inline website/integrations/services/aws/index.md:160]:
```suggestion
# This expression strips the default mapping from its 'photos' attribute,
```

Comment (tanberry) [reply website/integrations/services/aws/index.md:160]: @justSem please double-review this suggestion to make sure I did not change the technical accuracy when I changed the grammar. ;-)

File: website/integrations/services/aws/index.md [hunk 24]
```diff
@@ -75,3 +82,94 @@ To use the user's username, use this snippet
...
+# which is a forbidden property in AWS IAM.
+return {
+    "photos": None,
+}
+```
+
+-   Click _Save_. Navigate back to your SCIM provider, click _Edit_
+-   Under _User Property Mappings_ select the default mapping and the mapping you've just created.
```
Comment (tanberry) [inline website/integrations/services/aws/index.md:168]:
```suggestion
-   Under _User Property Mappings_ select the default mapping and the mapping that you just created.
```

File: website/integrations/services/aws/index.md [hunk 25]
```diff
@@ -75,3 +82,94 @@ To use the user's username, use this snippet
...
+-   Click _Update_
+
+-   Navigate to your application, click _Edit_.
+-   Under _Backchannel providers_ add the SCIM provider you've created.
```
Comment (tanberry) [inline website/integrations/services/aws/index.md:172]:
```suggestion
-   Under _Backchannel providers_ add the SCIM provider that you created.
```

File: website/integrations/services/aws/index.md [hunk 26]
```diff
@@ -75,3 +82,94 @@ To use the user's username, use this snippet
...
+-   Click _Update_
+
+The SCIM provider should sync automatically whenever you create/alter/remove anything. You can manually sync by going to your SCIM provider and click the _Run sync again_ button. Once it synced, you should see the users and groups in your AWS IAM center.
```
Comment (tanberry) [inline website/integrations/services/aws/index.md:175]:
```suggestion
The SCIM provider should sync automatically whenever you create/alter/remove anything. You can manually sync by going to your SCIM provider and click the _Run sync again_ button. Once the SCIM provider has synced, you should see the users and groups in your AWS IAM center.
```

**Review by tanberry — APPROVED** [review]
Great content, thank you @justSem! I made lots of copy-edits, please do let me know if any questions on them. Thanks so much for contributing to the docs!

File: website/integrations/services/aws/index.md (hunk 23 above)
Comment (justSem) [reply website/integrations/services/aws/index.md:160]: This is correct!

**Comment by justSem** [comment]
> Great content, thank you @justSem! I made lots of copy-edits, please do let me know if any questions on them. Thanks so much for contributing to the docs!

Heh, your edits make sense.
English is not my primary language so there will be the odd mistake here-and-there.

I tried batch-approving the suggestions, but for some reason github wouldn't let me. Not sure if I need any write-rights. Regardless, your edits look totally fine to me, thank you!

**Comment by tanberry** [comment]
@justSem if you go to the Files tab (not Conversation tab), can you click on Add to Batch for each suggestion? It's still a lot of clicking, sorry.

**Comment by justSem** [comment]
> @justSem if you go to the Files tab (not Conversation tab), can you click on Add to Batch for each suggestion? It's still a lot of clicking, sorry.

For some reason doing it all again worked this time... well, here you go :)

**Pull request merged**